client/app/components/queries/query-editor.js

```diff
@@ -119,31 +121,43 @@ function queryEditor(QuerySnippet, $timeout) {
 
         const schemaCompleter = {
           getCompletions(state, session, pos, prefix, callback) {
-            if (prefix.length === 0 || !$scope.schema) {
+            // make a variable for the auto completion in the query editor
+            $scope.autoCompleteSchema = $scope.schema; // removeExtraSchemaInfo(
+            if (prefix.length === 0 || !$scope.autoCompleteSchema) {
               callback(null, []);
               return;
             }
 
-            if (!$scope.schema.keywords) {
+            if (!$scope.autoCompleteSchema.keywords) {
               const keywords = {};
 
-              $scope.schema.forEach((table) => {
+              $scope.autoCompleteSchema.forEach((table) => {
                 keywords[table.name] = 'Table';
 
-                table.columns.forEach((c) => {
-                  keywords[c] = 'Column';
+                table.columns.forEach((c) => { // autoCompleteColumns
+                  if (c.charAt(c.length - 1) === ')') {
+                    let parensStartAt = c.indexOf('(') - 1;
+                    c = c.substring(0, parensStartAt);
+                    parensStartAt = 1; // linter complains without this line
+                  }
+                  // remove '[P] ' for partition keys
+                  if (c.charAt(0) === '[') {
+                    c = c.substring(4, c.length);
+                  }
+                  // keywords[c] = 'Column'; // dups columns
```

Comment:
Why the commented out line here?

Reply:
This is essentially my note to self that `keywords[c]` will duplicate columns in the list, that's why I had to make it `keywords[`${table.name}.${c}`]`. This will prevent me from trying to change it back later to reduce complexity and create a bug.